### PR TITLE
 refactor(app): refactor useRunStatuses and remove unnecessary logic

### LINKED ISSUE #11199: Refactor useRunStatuses and send module command logic
## Overview

There is a bunch of extra unnecessary logic throughout all the components that send module commands. Also the `useRunStatuses` hook does a lot and should probably be turned into a utility function. Clean this up.

## Implementation

- [ ] refactor `useRunStatuses` into a utility probably
- [ ] refactor to remove unnecessary logic in `useModuleOverflowMenu`, `useLatchControls`, temp, mag deck, TC, and H-S slideouts

## Acceptance Criteria

- [ ] should retrain full functionality despite the changes

### PR DESCRIPTION
**Pull request by jerader**
closes #11199 

# Overview

The logic used to send module controls when a run is terminal is overcomplicated, with the use of unnecessary logic. Additionally, the `useRunStatuses` hook is a bit confusing and unclear. This PR cleans up the logic and creates a new hook `useIsLegacySessionInProgress` so the stuff that `useRunStatuses` does makes more sense. Additionally, this fixes a bug found in `useModuleIdFromRun` where it white screens if you try to send a module command to a module not in the protocol but plugged in when the run is in terminal state.

# Changelog

- clean up logic in components that send a module command
- clean up `useRunStatuses` hook and create new hook `useIsLegacySessionInProgress`
- fix bug in `useModuleIdFromRun`

# Review requests

- smoke test sending commands when no run id is created, when run is idle, and when run is terminal to make sure this PR retains all functinoality.

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/Devices/hooks/useIsLegacySessionInProgress.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useAllSessionsQuery } from '@opentrons/react-api-client'
+
+export function useIsLegacySessionInProgress(): Boolean {
+  const allSessionsQueryResponse = useAllSessionsQuery()
+
+  const isLegacySessionInProgress =
+    allSessionsQueryResponse?.data?.data != null &&
+    allSessionsQueryResponse?.data?.data?.length !== 0
+
+  return isLegacySessionInProgress
```
Comment (koji) [inline app/src/organisms/Devices/hooks/useIsLegacySessionInProgress.ts]: totally optional

```ts
return allSessionsQueryResponse?.data?.data != null &&
    allSessionsQueryResponse?.data?.data?.length !== 0
```

File: app/src/organisms/Devices/hooks/useRunStatuses.ts [hunk 2]
```diff
@@ -25,17 +22,11 @@ export function useRunStatuses(): RunStatusesInfo {
...
-
-  const isLegacySessionInProgress =
-    allSessionsQueryResponse?.data?.data != null &&
-    allSessionsQueryResponse?.data?.data?.length !== 0
-
   const isRunStill =
     runStatus === RUN_STATUS_SUCCEEDED ||
     runStatus === RUN_STATUS_STOPPED ||
     runStatus === RUN_STATUS_FAILED ||
     runStatus === RUN_STATUS_IDLE
```
Comment (koji) [inline app/src/organisms/Devices/hooks/useRunStatuses.ts]: maybe
```suggestion
  const isRunStill = isRunTerminal || isRunIdle
```

Comment (jerader) [reply app/src/organisms/Devices/hooks/useRunStatuses.ts]: oooh i like this, thanks for the suggestion!

File: app/src/organisms/ModuleCard/useModuleIdFromRun.ts [hunk 3]
```diff
@@ -32,7 +32,12 @@ export function useModuleIdFromRun(
     attachedModule => attachedModule.serialNumber === module.serialNumber
   )
 
-  const moduleIdFromRun = loadModuleCommands?.[moduleIndex].result.moduleId
+  //  result is possibly undefined if you send a module command when run is terminal
+  //  to a module that is not in the protocol but plugged into the robot
+  const moduleIdFromRun =
+    loadModuleCommands?.[moduleIndex] != null
+      ? loadModuleCommands?.[moduleIndex].result.moduleId
+      : ''
```
Comment (koji) [inline app/src/organisms/ModuleCard/useModuleIdFromRun.ts:38]: small thing but in our code base, we use `""` for the string we use `''` with curly brackets? 

```suggestion
      : ""
```

Comment (jerader) [reply app/src/organisms/ModuleCard/useModuleIdFromRun.ts:38]: @koji  `prettier` changes this back to `''`

**Review by koji — APPROVED** [review]
The changes look good to me.
Left a couple of comments

File: app/src/organisms/ModuleCard/useModuleIdFromRun.ts (hunk 3 above)
Comment (koji) [reply app/src/organisms/ModuleCard/useModuleIdFromRun.ts:38]: I see.

File: app/src/organisms/ModuleCard/useModuleIdFromRun.ts [hunk 4]
```diff
@@ -32,7 +32,12 @@ export function useModuleIdFromRun(
     attachedModule => attachedModule.serialNumber === module.serialNumber
   )
 
-  const moduleIdFromRun = loadModuleCommands?.[moduleIndex].result.moduleId
+  //  result is possibly undefined if you send a module command when run is terminal
+  //  to a module that is not in the protocol but plugged into the robot
```
Comment (shlokamin) [inline app/src/organisms/ModuleCard/useModuleIdFromRun.ts]: i would either delete this comment or rephrase it to indicate that this could happen any time there are more modules of a certain type plugged in than the amount in the protocol

File: app/src/organisms/ModuleCard/useModuleIdFromRun.ts (hunk 3 above)
Comment (shlokamin) [inline app/src/organisms/ModuleCard/useModuleIdFromRun.ts:38]: i don't love this but can't think of a better way to do this without the hook returning an optional error, and having consumers deal with that error if it's present.

if we go the route of falling back to an empty string, we should console.error that there is no run specific module id associated with the attached module passed in the hook.

Comment (jerader) [reply app/src/organisms/ModuleCard/useModuleIdFromRun.ts:38]: that makes sense, thanks!
